### ISSUE TITLE
definitions of 3D structures?

### ISSUE THREAD

**Issue by mdsumner** [albz/misc3d_data_sharing#1]
Looking at this now, I only see matrices of the form `reshape(values, [Nz Nr])` and there's no problem reading these in with readBin (using similar code to Matlab's fread). Literally all of the code is "read the Nr, Nz, then read all the values from the stream and push into a matrix and transpose". 

But, I can't see how any of this defines a surface in 3D or a field that can be contoured for something in 3D, perhaps there's implicit relations between these variables? I'm happy to explore if there's something more.

**Comment by albz** [comment]
You are right, I have erroneously only uploaded the slices for dimension issues. The 3d structures are much larger. I need to check data and find a smart way to share them so you can have a look. I can upload here the python I am using to read the 3D matrixes but I need to find a way to share the 3D plots (each file is about 250 Mb each). If you have any suggestions I will be happy to share them. What about google drive? Would that be ok with you?

**Comment by mdsumner** [comment]
Sure, that's ok - it won't happen in a hurry,but I'm happy to explore if you are ;)

**Comment by mdsumner** [comment]
Oh, I didn't keep the code on hand but fyi the reading in a line like

```
fread(con, 1, 'float64')
```

has an equivalent in R of

```
readBin(con, n = 1, what = "numeric", size = 8)
```

So all those freads are easily (albeit tediously) converted. I don't know any .arch readers in R, but there might be some. 

P. S. I use *con* because the function to open a file connection in R is literally called *file*. 

```
con <- file("myfilename.etc", mode = "rb")

```

You can seek and so forth on that as usual.

**Comment by albz** [comment]
I agree some of the parts are just boring conversions. But it shouldn't take too long (I hope).

There are binary `integer` and `floating points`, so just need to be precise in the conversion.

I can work on the reader, not now, but I can give as soon as possible. And if you feel like exploring the data I can try to share it somehow. Just let me know what's the best way to let you check the binary output. Thanks!
